test/factories/prismaHook.factory.ts

```diff
@@ -0,0 +1,32 @@
+import { Factory } from 'fishery';
+import { createRandomAddress } from '../utils';
+import { Chain } from '@prisma/client';
+
+import { HookData } from '../../modules/sources/transformers/hook-transformer';
+
+class PrismaHookFactory extends Factory<HookData> {}
+
+export const hookFactory = PrismaHookFactory.define(({ params }) => {
+    const hookAddress = params?.address ?? createRandomAddress();
+
+    return {
+        address: hookAddress,
+        name: params.name || 'Test Hook',
+        enableHookAdjustedAmounts: params?.enableHookAdjustedAmounts ?? false,
+        shouldCallAfterSwap: params?.shouldCallAfterSwap ?? false,
+        shouldCallBeforeSwap: params?.shouldCallBeforeSwap ?? false,
```

Comment:
Factory works in a way that you can simply state the default values for each parameter.
They will be used only if the parameter was not overwritten on `params`.
In other words, you can do this:
```suggestion
    return {
        address: createRandomAddress(),
        name: 'Test Hook',
        enableHookAdjustedAmounts: false,
        shouldCallAfterSwap: false,
        shouldCallBeforeSwap: false,
        ...
```
and you'll have the same result.

Reply:
Thanks. I have incorporated your suggestion.